### PR TITLE
fix(vector) : fix append path to empty path and add map functions.

### PR DESCRIPTION
**Pull request by onecoolx**
Help us review this PR! Anyone can [approve it or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)

### CODE REVIEW

File: src/draw/lv_draw_vector.h [hunk 1]
```diff
@@ -205,6 +205,20 @@ void lv_matrix_skew(lv_matrix_t * matrix, float skew_x, float skew_y);
  */
 void lv_matrix_multiply(lv_matrix_t * matrix, const lv_matrix_t * matrix2);
 
+/**
+ * Transform the coordinates of a point using given matrix
+ * @param point            pointer to a point
+ * @param matrix           pointer to a matrix
+ */
+void lv_matrix_map_point(lv_fpoint_t * point, const lv_matrix_t * matrix);
```
Comment (kisvegabor) [inline src/draw/lv_draw_vector.h:213]: Is `map` a well known keyword here? `transform` sounds more common to me.

As the operation is exacuted on the point, I think it should be:
```c
void lv_point_???_matrix(lv_fpoint_t * point, const lv_matrix_t * matrix);
```

File: src/draw/lv_draw_vector.h [hunk 2]
```diff
@@ -269,6 +283,13 @@ void lv_vector_path_cubic_to(lv_vector_path_t * path, const lv_fpoint_t * p1, co
  */
 void lv_vector_path_close(lv_vector_path_t * path);
 
+/**
+ * Get the bounding box of a path
+ * @param area              pointer to a `lv_area_t` variable for bounding box
+ * @param path              pointer to a path
+ */
+void lv_vector_path_get_bounding(lv_area_t * area, const lv_vector_path_t * path);
```
Comment (kisvegabor) [inline src/draw/lv_draw_vector.h:291]: Same here

File: src/draw/lv_draw_vector.c [hunk 3]
```diff
@@ -438,12 +476,12 @@ void lv_vector_path_append_path(lv_vector_path_t * path, const lv_vector_path_t
     uint32_t npoint_size = lv_array_length(&subpath->points);
 
     lv_array_resize(&path->ops, ops_size + nops_size);
-    uint8_t * start_ops = lv_array_get(&path->ops, ops_size - 1) + sizeof(lv_vector_path_op_t);
+    uint8_t * start_ops = path->ops.size == 0 ? path->ops.data : lv_array_get(&path->ops, ops_size - 1) + sizeof(lv_vector_path_op_t);
```
Comment (XuNeo) [inline src/draw/lv_draw_vector.c:479]: It has been fixed here.
You can rebase to master now.
https://github.com/lvgl/lvgl/pull/5300/files#diff-1964caa407fbbf719ca3cd91996e37573dec510d6f48e88f8c22e1183fdc3915L433-R443

Comment (onecoolx) [reply src/draw/lv_draw_vector.c:479]: sure, I will fix it and reopen the pullrequest

File: src/draw/lv_draw_vector.h (hunk 1 above)
Comment (onecoolx) [reply src/draw/lv_draw_vector.h:213]: ok , let me think about it

Comment (onecoolx) [reply src/draw/lv_draw_vector.h:213]: how about  "lv_point_transform_by_matrix" ?

File: src/draw/lv_draw_vector.h (hunk 2 above)
Comment (onecoolx) [reply src/draw/lv_draw_vector.h:291]: that is getting the bounding box for a path

File: src/draw/lv_draw_vector.h (hunk 1 above)
Comment (kisvegabor) [reply src/draw/lv_draw_vector.h:213]: I like it!

File: src/draw/lv_draw_vector.h (hunk 2 above)
Comment (kisvegabor) [reply src/draw/lv_draw_vector.h:291]: I mean the variable on which the operation is executed should be the first.

```c
void lv_vector_path_get_bounding(const lv_vector_path_t * path, lv_area_t * area);
```

Comment (onecoolx) [reply src/draw/lv_draw_vector.h:291]: ok， got it

**Pull request closed without merging**